### PR TITLE
build-275: add CSI build source volume to build api

### PR DESCRIPTION
**Pull request by jkhelil**
https://issues.redhat.com/browse/BUILD-275

### CODE REVIEW

**Review by adambkaplan — CHANGES_REQUESTED** [review]
@jkhelil please run `make update` to ensure all appropriate functions are regenerated (example - deepcopy functions).

**Comment by adambkaplan** [comment]
/hold

We shouldn't merge until all the code is ready (openshift-apiserver, openshift-controller-manager, builder).

**Comment by adambkaplan** [comment]
@jkhelil looks like the deep copy functions did not update - are you sure you committed all changes from `make update`?

**Comment by adambkaplan** [comment]
@jkhelil I had success running `make generate-in-container`, which runs the make update commands inside a container that has all the tooling dependencies installed.

The make target defaults to podman as the container, if you are using docker you can run `make generate-in-container RUNTIME=docker`

**Comment by adambkaplan** [comment]
You also need to ensure that your `_output` directory is empty - I got bit by this in my first attempt.

File: build/v1/types.go [hunk 1]
```diff
@@ -1430,6 +1433,10 @@ type BuildVolumeSource struct {
 	// configMap represents a ConfigMap that should populate this volume
 	// +optional
 	ConfigMap *corev1.ConfigMapVolumeSource `json:"configMap,omitempty" protobuf:"bytes,3,opt,name=configMap"`
+
+	// CSI represents ephemeral storage that is handled by certain external CSI drivers
```
Comment (adambkaplan) [inline build/v1/types.go:1437]: I believe we use lowercase so that the generated OpenAPI doc matchs the yaml/json. Also adding a suggestion to improve the grammar:

```suggestion
	// csi represents ephemeral storage provided by external CSI drivers which support this capability
```

**Review by adambkaplan — CHANGES_REQUESTED** [review]
Cosmetic items on the godoc for the API, otherwise looks good to me.

**Review by adambkaplan — APPROVED** [review]
/lgtm

**Comment by adambkaplan** [comment]
bump @sttts

File: Makefile [hunk 2]
```diff
@@ -1,3 +1,4 @@
+SHELL:=/usr/bin/bash
```
Comment (bparees) [inline Makefile:1]: this should be a separate PR, if we want this change at all
